File: TwoSum/twosum.js
/**
 * @param {number[]} nums - The array of num
 * @param {number} target - The target sum
 * @returns {number[]} returns zero based index two num of which sum = target
 */
var twoSum = function (nums, target) {
	const hashMap = {};

	for (let i = 0; i < nums.length; i++) {
		let remain = target - nums[i];

		if (remain in hashMap) {
			return [hashMap[remain], i];
		} else {
			hashMap[nums[i]] = i;
		}
	}

	return [];
};

File: TwoSum/twosum.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    final hashMap = {};

    for (var i = 0; i < nums.length; i++) {
      int remain = target - nums[i];
      if (hashMap.containsKey(remain)) {
        return [hashMap[remain], i];
      } else {
        hashMap[nums[i]] = i;
      }
    }

    return [0, 0];
  }
}
